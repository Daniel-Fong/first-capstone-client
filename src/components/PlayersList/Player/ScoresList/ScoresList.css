.add-score-h2 {
    font-size: 1.2em;
}

.add-score-form input {
    display: block;
}

.scores-list-ul {
    width: 90%;
    margin-left: -20%;
}

.player-li {
    width: 25%;
    margin-right: 1em;
}

.score-li {
    margin-left: -2em;
    padding-top: .5em;
    padding-bottom: .5em;
}

.players-ul {
    margin-left: -2em;
}

.add-score-form {
    text-align: left;
    margin-left: 0;
}

.add-score-form input {
    width: 8em;
}

@media screen and (min-width: 700px) {
    .scores-list-ul {
        width: 85%;
    }

    .add-score-form {
        margin-left: 1em;
    }
}