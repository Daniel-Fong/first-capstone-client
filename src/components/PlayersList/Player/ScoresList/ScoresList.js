import React from 'react'
import Score from './Score/Score'
import config from '../../../../config'
import TokenService from '../../../../services/token-service'
import './ScoresList.css'
class ScoresList extends React.Component {
    state = {
        scores: [],
        score: '',
        note: '',
    }
    componentDidMount() {
        this.fetchScores()
            .then(scores => this.setState({ scores: scores }))
    }

    handleDeleteScore = (e, id) => {
        const scoreid = id
        return (
            fetch(`${config.API_ENDPOINT}/scores/${scoreid}`, {
                method: 'DELETE',
                headers: {
                    'authorization': `bearer ${TokenService.getAuthToken()}`
                },
            })
            .then(res => {
                if (!res.ok)
                    return res.json().then(e => Promise.reject(e))
                // return res.json()
            })
            .then((score) => {
              this.setState({ scores: this.state.scores.filter(score => score.id !== scoreid)})
            })
            .catch(error => {
              console.error({ error })
            })
        )}

    handleAddScore = (e) => {
        e.preventDefault()
        const score = e.target.score.value
        const note = e.target.note.value
        let newScore = JSON.stringify({
            playerid: this.props.playerid,
            gameid: this.props.gameid,
            score: score,
            note: note
        })
        return(
            fetch(`${config.API_ENDPOINT}/scores/${this.props.gameid}/${this.props.playerid}`, {
                method: 'POST',
                headers: {
                    'content-type': 'application/json',
                    'authorization': `bearer ${TokenService.getAuthToken()}`
                },
                body: newScore
            })
            .then(res => {
                if (!res.ok)
                  return res.json().then(e => Promise.reject(e))
                return res.json()
              })
              .then((score) => {
                this.setState({ scores: [...this.state.scores, score]})
              })
              .catch(error => {
                console.error({ error })
              })
        )}

    fetchScores() {
        return(
            fetch(`${config.API_ENDPOINT}/scores/${this.props.gameid}/${this.props.playerid}`, {
                headers: {
                  'authorization': `bearer ${TokenService.getAuthToken()}`,
                }
            })
              .then(res => {
                if (res.ok) {
                  return res.json();
                }
                return Promise.reject('Error fetching scores from server')
              })
              .catch(err => {
                this.setState({error: err})
              })
          )
        } 

    handleScoreChange(e) {
      const target = e.target;
      const value = target.value;

      this.setState({
        score: value
      })
    }

    handleNoteChange(e) {
      const target = e.target;
      const value = target.value;

      this.setState({
        note: value
      })
    }

    render() {
        return(
            <div className='scores-list-div'>
            <h3>Score(s)</h3>
            <ul className='scores-list-ul'>
                {this.state.scores.map(score => <Score key={score.id} handleDeleteScore={this.handleDeleteScore} score={score} />)}
            </ul>
            <form className='add-score-form' onSubmit={(e) => {
                e.preventDefault()
                this.handleAddScore(e)
                this.setState({
                  score: '',
                  note: '',
                })
                }}>
                <h2 className='add-score-h2'>Add New Score or Note</h2>
                <label htmlFor='score'>Score</label>
                <input name='score' value={this.state.score} onChange={(e) => {
                  this.handleScoreChange(e)
                }}/>
                <label htmlFor='note'>Note</label>
                <input name='note' value={this.state.note} onChange={(e) => {
                  this.handleNoteChange(e)
                }}/>
                <button className='add-score-button' type='submit'>Add</button>
            </form>
            </div>
        )
    }
}

export default ScoresList;