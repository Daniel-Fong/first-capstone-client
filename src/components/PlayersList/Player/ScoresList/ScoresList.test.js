import React from 'react'
import ReactDOM from 'react-dom';
import ScoresList from './ScoresList'
import renderer from 'react-test-renderer'

describe('<ScoresList />', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ScoresList />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it ('renders the ui as expected', () => {
        const mockScoresList= renderer
            .create(<ScoresList />)
            .toJSON();
        expect(mockScoresList).toMatchSnapshot();
    })
})