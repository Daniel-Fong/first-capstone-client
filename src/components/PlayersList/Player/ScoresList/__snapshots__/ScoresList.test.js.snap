// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoresList /> renders the ui as expected 1`] = `
<div
  className="scores-list-div"
>
  <h3>
    Score(s)
  </h3>
  <ul
    className="scores-list-ul"
  />
  <form
    className="add-score-form"
    onSubmit={[Function]}
  >
    <h2
      className="add-score-h2"
    >
      Add New Score or Note
    </h2>
    <label
      htmlFor="score"
    >
      Score
    </label>
    <input
      name="score"
      onChange={[Function]}
      value=""
    />
    <label
      htmlFor="note"
    >
      Note
    </label>
    <input
      name="note"
      onChange={[Function]}
      value=""
    />
    <button
      className="add-score-button"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
